### PR TITLE
Add support to load LOC from a mappings file

### PR DESCRIPTION
**Pull request by dnkoutso**
The `loc_json_file_path ` param can be used in conjunction with a `dot` graph style in order to use proportional LOC on each module.

### CODE REVIEW

File: uberpoet/buckprojectgen.py [hunk 1]
```diff
@@ -136,9 +146,9 @@ def gen_lib_module(self, module_node, loc_per_unit):
         buck_text = self.bzl_lib_template.format(module_node.name, deps, self.wmo_state)
 
         # Make Swift Text
-        file_count = (loc_per_unit * module_node.code_units) / self.swift_file_size_loc
+        file_count = (max(self.swift_file_size_loc, loc_per_unit) * module_node.code_units) / self.swift_file_size_loc
```
Comment (dnkoutso) [inline uberpoet/buckprojectgen.py:149]: its now possible that passing a loc file might generate way too little LOC for a module, this way taking the max it is guaranteed we always produce at least 1 file for such modules. Happy to discuss alternatives

File: uberpoet/dotreader.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def __init__(self, modules_filter=None):
             # We are trying to simulate a non-test app build and ignore non-code targets such as asset
             # catalogs and schemes. Each module currently gets a static amount of code created for it,
             # so non-code modules will add code to the total when they shouldn't if we didn't filter them out.
-            modules_filter = ['test', 'scheme', 'assetcatalog', 'resources', 'fixture', 'needle']
+            modules_filter = ['test', 'scheme', 'assetcatalog', 'resources', 'fixture', 'needle', 'assets']
```
Comment (dnkoutso) [inline uberpoet/dotreader.py:52]: we use this naming for various modules in our codebase, I guess its not so bad to include this here for everything?

File: uberpoet/buckprojectgen.py (hunk 1 above)
Comment (uber-mahyar2) [reply uberpoet/buckprojectgen.py:149]: I think that is reasonable since a file is pretty small

File: uberpoet/dotreader.py (hunk 2 above)
Comment (uber-mahyar2) [reply uberpoet/dotreader.py:52]: sgtm

File: README.md [hunk 3]
```diff
@@ -24,6 +24,8 @@ With a mac computer that can run macOS 10.13+, install all the dependencies belo
     * Install with: `xcode-select --install` / [The mac app store](https://itunes.apple.com/us/app/xcode/id497799835)
 * [BUCK for Xcode project generation](https://buckbuild.com/)
 	* [Install instructions](https://buckbuild.com/setup/getting_started.html)
+* [CocoaPods for Xcode project generation](https://cocoapods.org/)
```
Comment (uber-mahyar2) [inline README.md:27]: I would put something that says you only need one of these, unless you want to test project generation for both.

File: README.md [hunk 4]
```diff
@@ -65,6 +67,39 @@ caffeinate -s pipenv run \
                 --app_gen_output_dir "$HOME/Desktop/multisuite_build_results/app_gen"
 ```
 
+You may also generate a project that matches your own project's dependency graph by using `--gen_type dot` parameter as well as supplying the location of the `dot` file that represents the graph:
```
Comment (uber-mahyar2) [inline README.md:70]: Could you link to the dot file format in wikipedia? https://en.wikipedia.org/wiki/DOT_(graph_description_language)

Also give an example command from buck (and cocoapods if you have it) to generate the dot graph? `buck query \"deps(target)\" --dot > file.gv`

File: README.md [hunk 5]
```diff
@@ -65,6 +67,39 @@ caffeinate -s pipenv run \
...
+```bash
+pipenv run ./genproj.py --output_directory "$HOME/Desktop/mockapp" \
+                        --project_generator_type "cocoapods" \
+                        --gen_type dot \
+                        --dot_file_path "$HOME/MyProject/my_project_graph.dot" \
+                        --dot_root_node_name "MyProject" \
+                        --lines_of_code 150000
+```
+
+Additionally, you may also supply an optional JSON file to be used as a LOC map.  This allows you to generate a project from your own dependency graph in which each generated module has proportional LOC to your original graph.
```
Comment (uber-mahyar2) [inline README.md:81]: Could you give an example `cloc` or `tokei` command on how to do so here?  Or did you do something else to generate the json file?

**Review by uber-mahyar2 — APPROVED** [review]
lgtm other than a few improvements to the readme

File: README.md (hunk 5 above)
Comment (dnkoutso) [reply README.md:81]: will do

**Pull request merged**